src/pathfind/astarsearch.cpp

```diff
@@ -84,26 +84,28 @@ struct node {
 		if (teleports && !teleports->empty()) {
 
 			double new_srch = 1.0;
-			auto sources = teleports->get_sources();
+			const auto& sources = teleports->get_sources();
 
 			std::set<map_location>::const_iterator it = sources.begin();
 			for(; it != sources.end(); ++it) {
 				const double tmp_srch = heuristic(c, *it);
 				if (tmp_srch < new_srch) { new_srch = tmp_srch; }
 			}
 
-			double new_dsth = 1.0;
-			auto targets = teleports->get_targets();
+			if(new_srch + 1.0 < h) {
```

Comment:
What does this condition mean?

Reply:
new_srch was the heuristic distance from the current position to the nearest teleport source. If new_srch + 1 was greater than the heuristic distance to the target there was no point in running the contents of the if. In the latest commit this part was moved before the path finding loop so it would be only ran once.